core/user_api.php

```diff
@@ -304,17 +309,18 @@ function user_ensure_email_unique( $p_email, $p_user_id = null ) {
 	}
 
 	if( !user_is_email_unique( $p_email, $p_user_id ) ) {
-		trigger_error( ERROR_USER_EMAIL_NOT_UNIQUE, ERROR );
+		throw new ClientException(
+			sprintf( "Email '%s' already used.", $p_email ),
+			ERROR_USER_EMAIL_NOT_UNIQUE );
 	}
 }
 
 /**
- * Check if the realname is a valid username (does not account for uniqueness)
- * Return 0 if it is invalid, The number of matches + 1
+ * Check if realname is specified, a unique real name, and doesn't match a username of another user.
  *
  * @param string $p_username The username to check.
  * @param string $p_realname The realname to check.
- * @return integer
+ * @return integer 0 realname matches another user's username, 1 no conflicts, >1 realname not unique.
```

Comment:
I know this is not a new behavior, but we cannot have two accounts with "John Smith" as a real name?

Reply:
Yep we can't.  It is a defensive thing to avoid confusion by users who may assign access rights and issues to the wrong `John Smith` and show real name is set to ON.